compact.go

```diff
@@ -757,16 +757,22 @@ func (c *LeveledCompactor) populateBlock(blocks []BlockReader, meta *BlockMeta,
 		}
 
 		for i, chk := range chks {
-			if chk.MinTime < meta.MinTime || chk.MaxTime > meta.MaxTime {
+			if chk.MinTime < meta.MinTime {
 				return errors.Errorf("found chunk with minTime: %d maxTime: %d outside of compacted minTime: %d maxTime: %d",
 					chk.MinTime, chk.MaxTime, meta.MinTime, meta.MaxTime)
 			}
 
+			// Block range is max exclusive, chunk min and max are inclusive, so chk.max == meta.max is wrong as well.
+			if chk.MaxTime >= meta.MaxTime {
```

Comment:
This loses us the sanity check for on-disk blocks.

A comment as to that this is only for the head block would also be good.

Reply:
> This loses us the sanity check for on-disk blocks.

what do you mean by this?

Reply:
@brian-brazil means that we want to rewrite chunks only for head chunks. 

For any other chunks this sanity check would be nice to have as it's an error case in any other case than head chunk.